.github/workflows/ci.yml

```diff
@@ -34,4 +34,4 @@ jobs:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
       - name: Test Site
-        run: bash tools/deploy.sh --dry-run
```

Comment:
```yaml
jobs:
  build:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        ruby: [2.5, 2.6, 2.7, 3]

    steps:
      - name: Checkout
        uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Setup Pages
        id: pages
        uses: actions/configure-pages@v1

      - name: Setup Ruby
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ matrix.ruby }}
          bundler-cache: true

      - name: Build site
        run: bundle exec jekyll b -d "_site${{ steps.pages.outputs.base_path }}"
        env:
          JEKYLL_ENV: "production"

      - name: Test site
        run: |
          bundle exec htmlproofer _site --disable-external --check-html --allow_hash_href
```

Reply:
The `jobs` of CI should be change to this.

Reply:
The `test.sh` script runs the same as the `Test site` step described above. I thought it better to re-use that.

Reply:
The decoupling of local development scripts and CI workflows makes the project much cleaner. In fact, for now, `test.sh` is a non-essential script. :)